lib/azure/armrest/storage_account_service.rb

```diff
@@ -292,7 +292,21 @@ def get_private_images(storage_accounts)
             next unless File.extname(blob.name).casecmp('.vhd') == 0
             next unless blob.properties.lease_state.casecmp('available') == 0
 
-            blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
+            max_retries = 0
+
+            begin
+              blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
+            rescue Errno::ECONNREFUSED, Azure::Armrest::TimeoutError => err
+              if max_retries < 3
+                max_retries += 1
+                sleep 1
```

Comment:
It only seems to flake out very briefly in my experience, so yes. And, if not, it will try again.